miditoolkit/midi/containers.py

```diff
@@ -317,46 +310,75 @@ def remove_invalid_notes(self, verbose=True):
 
     def __repr__(self):
         return 'Instrument(program={}, is_drum={}, name="{}")'.format(
-            self.program, self.is_drum, self.name.replace('"', r'\"'))
+            self.program, self.is_drum, self.name.replace('"', r"\"")
+        )
+
+    def __eq__(self, other):
+        # Here we check all tracks attributes except the name.
+        # The list attributes will be checked sequentially one by one, this means that
+        # if two Instrument objects have the same musical content, but with some elements in
+        # different orders (for example two notes with swapped indices in a list), the method will
+        # return False. To make this method insensible to the lists orders, you can manually sort
+        # them before calling it: `track.notes.sort(key=lambda x: (x.start, x.pitch, x.end, x.velocity))`.
+        # The same can be done for control_changes, pitch_bends and pedals.
+        if self.is_drum != other.is_drum or self.program != other.program:
+            return False
+
+        # Check list attributes.
+        # These list should all contain objects that is either a dataclass or implements `__eq__`.
+        lists_attr = [name for name, val in vars(self).items() if isinstance(val, list)]
+        for list_attr in lists_attr:
+            if len(getattr(self, list_attr)) != len(getattr(other, list_attr)):
+                return False
+            if any(a1 != a2 for a1, a2 in zip(getattr(self, list_attr), getattr(other, list_attr))):
+                return False
```

Comment:
What we do here is instead of checking each list attribute one by one, we automatically identify them and test them, for more concise code and lower maintenance cost